packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -88,6 +95,38 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
     static Texture: Constructor<Texture>;
     static Image: Constructor<Image>;
 
+    private[$provideOnUpdate](): () => void {
```

Comment:
Why do we need to wrap this function in a provider? A function is already an object you can pass if you simply pass `onUpdate` instead of `onUpdate()`. Or is there some problem with the meaning of `this`?

Reply:
So this was to prevent repeating the same code in different methods. Since I needed to create the `onUpdate` function the same way that was being done in the `updated` method I just pulled it out and wrapped to prevent it from having to be maintained in multiple places. I could just make it method that lives on `scene-graph` is that what you mean?

Reply:
I just turned it into a private method, this seems to work fine, not seeing any problems with the `this` reference

Reply:
Actually, looking at it now, I see even the `$onUpdate` wrapper is unnecessary; we can just pass `$needsRender` directly.